edge-hub/core/src/Microsoft.Azure.Devices.Edge.Hub.MqttBrokerAdapter/handlers/ConnectionHandler.cs

```diff
@@ -94,6 +99,20 @@ public async Task<bool> HandleAsync(MqttPublishInfo publishInfo)
             return false;
         }
 
+        public async Task CloseConnectionAsync(IIdentity identity)
+        {
+            var identityFound = false;
+            using (await this.guard.LockAsync())
```

Comment:
We need to lock this because something might get added to knownConnections?

Reply:
What race conditions we are trying to prevent with the lock? knownConnections is a ConcurrentDictionary already

Reply:
The story of the lock is that when the code updates the 'knowConnection', that needs a lock anyway (because it does multistep things. so a parallel call cannot mess it up) - so it would not be necessary using a concurrent dictionary (although I ended up using one, because it has a member normal dictionaries don't) And because of the lock on writes, I felt it is better to keep it symmetric, and lock the reads, too.   But yes, with the current ConcurrentDictionary the read operation would not be dangerous doing without a lock.